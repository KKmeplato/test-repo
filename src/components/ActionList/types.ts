import { DropdownMenuProps } from "@radix-ui/react-dropdown-menu";
import { ComponentPropsWithoutRef } from "react";

import { Variant } from "lib/types";

export interface ActionItemProps
  extends Omit<ComponentPropsWithoutRef<"button">, "onClick"> {
  active?: boolean;
  variant?: Extract<Variant, "critical">;
  prefixIcon?: React.ReactNode;
  suffixIcon?: React.ReactNode;
  sideHelpText?: string | React.ReactNode;
  bottomHelpText?: string | React.ReactNode;
  onClick?: () => void;
}

export interface ActionListSection {
  title?: string;
  items: ActionItemProps[];
}

export interface ActionListProps
  extends Pick<DropdownMenuProps, "onOpenChange"> {
  items?: ActionItemProps[];
  sections?: ActionListSection[];
  activator?: React.ReactNode;
}
