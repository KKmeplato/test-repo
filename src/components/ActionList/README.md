### ActionList

The Action List component provides users with a context-sensitive list of possible actions they can take on a given item. This list is typically revealed through interaction, offering a dropdown or context-menu-like appearance.

#### When to use

* When users need to perform actions on specific items or content
* To provide a consistent set of actions across multiple parts of the application
* When screen real estate is limited, and actions cannot be presented directly next to each item

```tsx
const items = [
  {
    id: "sort-asc",
    content: "Sort ascending",
    helpText: "You can also use keyboard controls",
    children: "Sort ascending",
    prefixIcon: <Icon.info />,
    onClick: () => console.log("Sort ascending clicked"),
    active: true,
  },
  {
    id: "sort-desc",
    content: "Sort descending",
    suffixIcon: <Icon.info />,
    onClick: () => console.log("Sort ascending clicked"),
  }
];

<ActionList
  items={items}
  activator={
    <Button
      variant="primary"
      disclosure
      onClick={() => console.log("Item click")}
    >
      Action List
    </Button>
  }
/> 
```

- [x] Meplato light theme
- [x] Meplato dark theme
- [x] Webridge light theme
- [x] Webridge dark theme