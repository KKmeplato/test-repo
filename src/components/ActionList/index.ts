export * from "./ActionList";
export * from "./types";
