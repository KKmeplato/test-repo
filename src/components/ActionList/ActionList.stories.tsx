import { action } from "@storybook/addon-actions";
import { Meta, StoryObj } from "@storybook/react";

import { ActionList } from "components/ActionList";
import { Button } from "components/Button";
import { Icon } from "components/icons";

import { ActionItemProps, ActionListProps } from "./types";

const meta: Meta<typeof ActionList> = {
  component: ActionList
};

export default meta;
type Story = StoryObj<typeof ActionList>;

const activator = (
  <Button variant="primary" disclosure="down" accessibilityLabel="">
    Open
  </Button>
);

const items: ActionItemProps[] = [
  {
    id: "sort-asc",
    children: "Sort ascending",
    sideHelpText: "XX",
    bottomHelpText: "You can also use keyboard controls",
    prefixIcon: <Icon.info />,
    onClick: action("sort ascending"),
    active: true
  },
  {
    id: "sort-desc",
    children: "Sort descending",
    suffixIcon: <Icon.info />,
    onClick: action("sort descending")
  },
  {
    id: "sort-disabled",
    children: "Sort disabled",
    disabled: true,
    prefixIcon: <Icon.info />,
    onClick: action("noop")
  },
  {
    id: "sort-critcal",
    children: "Sort critical disabled",
    disabled: true,
    suffixIcon: <Icon.info />,
    variant: "critical",
    onClick: action("sort critical")
  },
  {
    id: "sort-critcal-active",
    children: "Sort critical",
    active: true,
    suffixIcon: <Icon.info />,
    variant: "critical",
    onClick: action("sort critical active")
  }
];

const defaultItems = items;

const createStory = ({
  items = defaultItems,
  sections,
  activator
}: Partial<ActionListProps>): Story => ({
  args: {
    items: sections ? undefined : items,
    sections,
    activator
  }
});

export const WithActivator = createStory({
  activator: activator
});

export const WithSections = createStory({
  sections: [
    {
      title: "Sort options",
      items
    },
    {
      title: "Bulk actions",
      items
    }
  ]
});

export const CustomElement = createStory({
  items: [
    {
      id: "sort-asc",
      children: (
        <button className="text-lg text-primary">Custom element</button>
      ),
      onClick: action("Click custom element")
    },
    {
      id: "sort-desc",
      children: "Sort descending",
      prefixIcon: <Icon.info />,
      onClick: action("sort descending")
    }
  ]
});

const figmaExampleItems = [
  {
    children: "Import",
    onClick: action("import")
  },
  {
    children: "Export",
    onClick: action("export")
  },
  {
    children: "Duplicate",
    onClick: action("duplicate")
  },
  {
    children: "Archive",
    active: true,
    onClick: action("archive")
  }
];

export const FigmaExample = createStory({
  sections: [
    {
      items: figmaExampleItems
    },
    {
      title: "Section title",
      items: [
        {
          children: "Archive",
          variant: "critical",
          onClick: action("archive")
        }
      ]
    }
  ]
});
