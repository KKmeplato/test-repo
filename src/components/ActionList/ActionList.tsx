import * as DropdownMenu from "@radix-ui/react-dropdown-menu";
import { useState } from "react";

import { cn } from "lib/utils";

import { ActionItem } from "./components";
import { ActionListProps } from "./types";

export const ActionList = ({
  items,
  sections,
  activator,
  onOpenChange
}: ActionListProps) => {
  const [, setOpen] = useState(false);
  const shouldRenderAsDropdown = Boolean(activator);

  const getItemsMarkup = (items: ActionListProps["items"]) =>
    items?.map(({ onClick, ...item }, index) =>
      shouldRenderAsDropdown ? (
        <DropdownMenu.Item asChild key={item?.id || index} onSelect={onClick}>
          <ActionItem {...item} />
        </DropdownMenu.Item>
      ) : (
        <ActionItem key={item.id || index} {...item} onClick={onClick} />
      )
    );

  const itemsMarkup = items?.length ? getItemsMarkup(items) : null;

  const sectionsMarkup = sections?.length ? (
    <div className="group divide-y">
      {sections.map((section, index) => (
        <div
          key={index}
          className="group/section-item [&:not(:first-child)]:mt-1"
        >
          {section.title && (
            <p className="mx-4 mt-1 pb-2 pt-4 text-xs font-semibold group-first/section-item:pt-1">
              {section.title}
            </p>
          )}
          {getItemsMarkup(section.items)}
        </div>
      ))}
    </div>
  ) : null;
  const className =
    "flex flex-col overflow-hidden rounded-level-1 bg-surface py-2 shadow-md data-[state=closed]:animate-dropdown-close data-[state=open]:animate-dropdown-open max-w-max";

  const handleOpenChange = (open: boolean) => {
    setOpen(open);
    onOpenChange?.(open);
  };

  return shouldRenderAsDropdown ? (
    <DropdownMenu.Root onOpenChange={handleOpenChange} modal={false}>
      <DropdownMenu.Trigger asChild>{activator}</DropdownMenu.Trigger>
      <DropdownMenu.Portal container={document.getElementById("uikit-theme")}>
        <DropdownMenu.Content
          side="bottom"
          className={cn(className, "z-dropdown mt-1")}
          align="start"
        >
          {itemsMarkup || sectionsMarkup}
        </DropdownMenu.Content>
      </DropdownMenu.Portal>
    </DropdownMenu.Root>
  ) : (
    <div className={className}>{itemsMarkup || sectionsMarkup}</div>
  );
};

ActionList.displayName = "ActionList";
