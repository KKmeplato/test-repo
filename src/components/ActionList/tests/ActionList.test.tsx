import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { expect } from "vitest";

import { ActionList } from "../ActionList";

const mockClick = vi.fn();

describe("<ActionList />", () => {
  test("renders a simple ActionList", async () => {
    const items = [
      {
        id: "sort-asc",
        children: "Sort ascending",
        onClick: mockClick("sort-asc")
      },
      {
        id: "sort-desc",
        children: "Sort descending",
        onClick: mockClick("sort-desc")
      }
    ];
    const result = render(<ActionList items={items} />);

    expect(result.getByText("Sort ascending")).toBeDefined();
    expect(result.getByText("Sort descending")).toBeDefined();

    expect(result).toMatchSnapshot();

    await userEvent.click(result.getAllByRole("button")[0]);

    expect(mockClick).toHaveBeenCalledWith("sort-asc");
  });

  test("renders an ActionList with sections", async () => {
    const mockClick = vi.fn();

    const sections = [
      {
        title: "Sort options",
        items: [
          {
            id: "sort-asc",
            children: "Sort ascending",
            onClick: () => mockClick("sort-asc")
          },
          {
            id: "sort-desc",
            children: "Sort descending",
            onClick: () => mockClick("sort-desc")
          }
        ]
      },
      {
        title: "Bulk options",
        items: [
          {
            id: "edit",
            children: "Edit",
            onClick: () => mockClick("edit")
          },
          {
            id: "remove",
            children: "Sort descending",
            onClick: () => mockClick("remove")
          }
        ]
      }
    ];
    const result = render(<ActionList sections={sections} />);

    expect(result.getByText("Sort options")).toBeDefined();
    expect(result.getByText("Bulk options")).toBeDefined();
    expect(result).toMatchSnapshot();

    await userEvent.click(result.getAllByRole("button")[2]);

    expect(mockClick).toHaveBeenCalledWith("edit");
  });

  test("renders an ActionList as a dropdown", async () => {
    function Wrapper({ children }: { children: React.ReactElement }) {
      return <div id="uikit-theme">{children}</div>;
    }

    const items = [
      {
        id: "sort-asc",
        children: "Sort ascending",
        onClick: mockClick("sort-asc")
      },
      {
        id: "sort-desc",
        children: "Sort descending",
        onClick: mockClick("sort-desc")
      }
    ];
    const activator = <button>Open</button>;
    const result = render(<ActionList items={items} activator={activator} />, {
      wrapper: Wrapper
    });

    expect(result).toMatchSnapshot();

    await userEvent.click(result.getByText("Open"));

    expect(result.getAllByRole("menuitem")).toHaveLength(2);
    expect(result.getAllByRole("menuitem")[0]).toHaveTextContent(
      "Sort ascending"
    );
  });
});
