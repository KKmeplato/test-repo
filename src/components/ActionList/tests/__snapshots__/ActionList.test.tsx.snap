// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ActionList /> > renders a simple ActionList 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col overflow-hidden rounded-level-1 bg-surface py-2 shadow-md data-[state=closed]:animate-dropdown-close data-[state=open]:animate-dropdown-open max-w-max"
      >
        <div
          class="relative px-2"
        >
          <button
            class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
            id="sort-asc"
          >
            <span
              class="flex gap-4"
            >
              <span
                class="flex flex-col items-start text-sm font-normal"
              >
                Sort ascending
              </span>
            </span>
          </button>
        </div>
        <div
          class="relative px-2"
        >
          <button
            class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
            id="sort-desc"
          >
            <span
              class="flex gap-4"
            >
              <span
                class="flex flex-col items-start text-sm font-normal"
              >
                Sort descending
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col overflow-hidden rounded-level-1 bg-surface py-2 shadow-md data-[state=closed]:animate-dropdown-close data-[state=open]:animate-dropdown-open max-w-max"
    >
      <div
        class="relative px-2"
      >
        <button
          class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
          id="sort-asc"
        >
          <span
            class="flex gap-4"
          >
            <span
              class="flex flex-col items-start text-sm font-normal"
            >
              Sort ascending
            </span>
          </span>
        </button>
      </div>
      <div
        class="relative px-2"
      >
        <button
          class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
          id="sort-desc"
        >
          <span
            class="flex gap-4"
          >
            <span
              class="flex flex-col items-start text-sm font-normal"
            >
              Sort descending
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ActionList /> > renders an ActionList as a dropdown 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="uikit-theme"
      >
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          Open
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="uikit-theme"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        data-state="closed"
        id="radix-:r0:"
        type="button"
      >
        Open
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ActionList /> > renders an ActionList with sections 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col overflow-hidden rounded-level-1 bg-surface py-2 shadow-md data-[state=closed]:animate-dropdown-close data-[state=open]:animate-dropdown-open max-w-max"
      >
        <div
          class="group divide-y"
        >
          <div
            class="group/section-item [&:not(:first-child)]:mt-1"
          >
            <p
              class="mx-4 mt-1 pb-2 pt-4 text-xs font-semibold group-first/section-item:pt-1"
            >
              Sort options
            </p>
            <div
              class="relative px-2"
            >
              <button
                class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
                id="sort-asc"
              >
                <span
                  class="flex gap-4"
                >
                  <span
                    class="flex flex-col items-start text-sm font-normal"
                  >
                    Sort ascending
                  </span>
                </span>
              </button>
            </div>
            <div
              class="relative px-2"
            >
              <button
                class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
                id="sort-desc"
              >
                <span
                  class="flex gap-4"
                >
                  <span
                    class="flex flex-col items-start text-sm font-normal"
                  >
                    Sort descending
                  </span>
                </span>
              </button>
            </div>
          </div>
          <div
            class="group/section-item [&:not(:first-child)]:mt-1"
          >
            <p
              class="mx-4 mt-1 pb-2 pt-4 text-xs font-semibold group-first/section-item:pt-1"
            >
              Bulk options
            </p>
            <div
              class="relative px-2"
            >
              <button
                class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
                id="edit"
              >
                <span
                  class="flex gap-4"
                >
                  <span
                    class="flex flex-col items-start text-sm font-normal"
                  >
                    Edit
                  </span>
                </span>
              </button>
            </div>
            <div
              class="relative px-2"
            >
              <button
                class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
                id="remove"
              >
                <span
                  class="flex gap-4"
                >
                  <span
                    class="flex flex-col items-start text-sm font-normal"
                  >
                    Sort descending
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col overflow-hidden rounded-level-1 bg-surface py-2 shadow-md data-[state=closed]:animate-dropdown-close data-[state=open]:animate-dropdown-open max-w-max"
    >
      <div
        class="group divide-y"
      >
        <div
          class="group/section-item [&:not(:first-child)]:mt-1"
        >
          <p
            class="mx-4 mt-1 pb-2 pt-4 text-xs font-semibold group-first/section-item:pt-1"
          >
            Sort options
          </p>
          <div
            class="relative px-2"
          >
            <button
              class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
              id="sort-asc"
            >
              <span
                class="flex gap-4"
              >
                <span
                  class="flex flex-col items-start text-sm font-normal"
                >
                  Sort ascending
                </span>
              </span>
            </button>
          </div>
          <div
            class="relative px-2"
          >
            <button
              class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
              id="sort-desc"
            >
              <span
                class="flex gap-4"
              >
                <span
                  class="flex flex-col items-start text-sm font-normal"
                >
                  Sort descending
                </span>
              </span>
            </button>
          </div>
        </div>
        <div
          class="group/section-item [&:not(:first-child)]:mt-1"
        >
          <p
            class="mx-4 mt-1 pb-2 pt-4 text-xs font-semibold group-first/section-item:pt-1"
          >
            Bulk options
          </p>
          <div
            class="relative px-2"
          >
            <button
              class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
              id="edit"
            >
              <span
                class="flex gap-4"
              >
                <span
                  class="flex flex-col items-start text-sm font-normal"
                >
                  Edit
                </span>
              </span>
            </button>
          </div>
          <div
            class="relative px-2"
          >
            <button
              class="flex group w-full gap-4 rounded-lg px-2 py-2.5 focus-visible:outline focus-visible:outline-1 focus-visible:outline-focus focus-visible:ring-1 focus-visible:ring-focus focus-visible:ring-offset-1 disabled:text-disabled justify-between relative focus-visible:z-10 hover:z-10 text enabled:hover:bg-surface-hovered enabled:active:bg-surface-pressed items-center"
              id="remove"
            >
              <span
                class="flex gap-4"
              >
                <span
                  class="flex flex-col items-start text-sm font-normal"
                >
                  Sort descending
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
